vm/compiler/src/ledger/block/mod.rs

```diff
@@ -104,6 +116,11 @@ impl<N: Network> Block<N> {
         self.previous_hash
     }
 
+    /// Returns the coinbase proof.
+    pub const fn coinbase_proof(&self) -> &Option<CoinbaseSolution<N>> {
+        &self.coinbase_proof
```

Comment:
```suggestion
    pub const fn coinbase_proof(&self) -> Option<&CoinbaseSolution<N>> {
        self.coinbase_proof.as_ref()
```
unless `CoinbaseSolution` is `Copy`, in which case just `self.coinbase_proof`